Update all dev dependencies. Check in audit fix results

richardgirges opened the pull request:
* I've updated all dependencies to the latest versions.
* Moved `md5` out of `devDependencies` and into main `dependencies` since it's a required package
* Eliminated support in Travis CI for Node 6 and 7 (we need to make a major or minor release because of this)
* Added support for Node 11 and 12 in Travis CI
* Lastly, checked in results of `npm audit fix` in package-lock.json. 

NOTE: See my comments below, there is one test failing in Node 8 and 9.

NOTE 2: There are still various security alerts that we need to look into, I believe most of them are related to sub-dependencies of the `istanbul` package.

richardgirges commented:
I've resolved a couple more issues. 

1. Now we only support Node v8 - 12 (support for Node 6 and 7 is over)
2. I re-integrated Travis CI when PRs are open

I've noticed that there is one test failing in Node 8 and 9. Related to the `checkAndMakeDir` function. Can you take a look at this @RomanBurunkov ?

RomanBurunkov commented:
Hi @richardgirges ,

According to the md5 module.
It used only for testing.
Could u please clarify the reason moving it from dev deps?

richardgirges commented:
@RomanBurunkov upon inspecting the code, I see that we're requiring the `md5` module in tempFileHandler.js and memHandler.js - both of which are used in the primary package. We also return `req.files.<filename>.md5` as a property (one fo the core features of this library).

As for Node 8 and 9, maybe it's better to deprecate support for those versions of Node and only support Node 10+ so that we don't have to add new dependencies to the project. What do you think?

RomanBurunkov commented:
> @RomanBurunkov upon inspecting the code, I see that we're requiring the `md5` module in tempFileHandler.js and memHandler.js - both of which are used in the primary package. We also return `req.files.<filename>.md5` as a property (one fo the core features of this library).
> 
> As for Node 8 and 9, maybe it's better to deprecate support for those versions of Node and only support Node 10+ so that we don't have to add new dependencies to the project. What do you think?



There are no md5 requirements in those files.
Both of them uses native node module 'crypto' to calculate md5.

`const crypto = require('crypto');`

`const hash = crypto.createHash('md5');`

richardgirges commented:
Ah you're right @RomanBurunkov - I misread the import statements. I switched md5 package back to devDependencies.

I also removed Node 8 and 9 from the TravisCI tests. 

When you do the next release we'll just need to put a note about breaking change for not supporting Node 8 and 9 anymore
